### PR TITLE
add system-arch-dracut

### PR DESCRIPTION
**Pull request by ojafuenf**
also make kexec configurable

### CODE REVIEW

File: setup-server-os [hunk 1]
```diff
@@ -129,6 +131,9 @@ elif 'system-gentoo' in cfg.modules:
 
     if 'nbd' in cfg.modules:
         install_folder('overlays/server-os-dracut-nbd')
+        if 'system-arch-dracut' in cfg.modules:
+            install_folder('overlays/server-os-dracut-arch')
+        command('dracut', '/boot/initramfs-stiefel.img', '--add', ' nbd ', '--no-hostonly', '--force')
```
Comment (TheJJ) [inline setup-server-os]: the initramfs should not be generated upon server os setup

File: test-qemu [hunk 2]
```diff
@@ -389,7 +390,7 @@ with contextlib.ExitStack() as exit_stack:
         "qemu-system-x86_64",
         "-machine", "q35,accel=kvm",
         "-cpu", "host",
-        "-m", str(3 * 1024),
+        "-m", str(3 * 1024 if args.mode == "server" else 5 * 1024),
```
Comment (TheJJ) [inline test-qemu:393]: we really need to investigate this...

**Review by TheJJ — CHANGES_REQUESTED** [review]
apart from these things should be good to go (didn't test arch-dracut though)

File: setup-server-os (hunk 1 above)
Comment (ojafuenf) [reply setup-server-os]: Why not? Otherwise the user has to generate it manually.

But we probably should use cfg.boot.load.initrd as the target file.

Comment (TheJJ) [reply setup-server-os]: the regular system initrd should support nbd then, having it two times just wastes space. also the output path is hardcoded here without any info on how to process it further (how that file is actually used for booting then)

**Review by TheJJ — APPROVED** [review]

**Pull request merged**